src/misc.jl

```diff
@@ -60,7 +60,7 @@ run lengths.
 function inverse_rle(vals::AbstractVector{T}, lens::AbstractVector{<:Integer}) where T
     m = length(vals)
     length(lens) == m || raise_dimerror()
-
+    all(>=(0), lens) || throw(ArgumentError("lengths must be non-negative"))
     r = Vector{T}(undef, sum(lens))
     p = 0
     @inbounds for i = 1 : m
```

Comment:
How about doing `j >= 0 || throw(ArgumentError("lengths must be positive"))` here instead, to avoid going over the vector twice?

Reply:
I guess the tradeoff is iterate twice or iterate once but always allocate a (potentially large) array and branch at each iteration. I suspect iterating twice would be more efficient but I haven't actually benchmarked or anything like that.

Reply:
Allocating before throwing an error doesn't  sound like a problem to me as it shouldn't happen in valid code. As for branching on each iteration, given that there's already a branch I imagine it won't make the loop slower. But not a big deal anyway.

Reply:
> Allocating before throwing an error doesn't sound like a problem to me as it shouldn't happen in valid code.

Ha yeah, fair point.

> As for branching on each iteration, given that there's already a branch I imagine it won't make the loop slower.

Oh you're right, for some reason the `while` loop didn't register to me as branching.

I'm happy to change if you prefer, I don't really care either way.

Reply:
Maybe change it if you feel like it, otherwise it's fine.